File: CS/PrismOnDXDocking.Infrastructure/RegionBehaviors/LayoutGroupRegionBehavior.cs
using System.ComponentModel.Composition;
using System.Windows;
using DevExpress.Xpf.Docking;
using Prism.Regions;
using Prism.Regions.Behaviors;

namespace PrismOnDXDocking.Infrastructure.Behaviors {
    [Export(typeof(LayoutGroupRegionBehavior)), PartCreationPolicy(CreationPolicy.NonShared)]
	public class LayoutGroupRegionBehavior : RegionBehavior, IHostAwareRegionBehavior {
		[Import]
		public IRegionManager RegionManager { get; set; }
        public DependencyObject HostControl { get; set; }
        protected override void OnAttach() {
			RegisterRegion();
		}
		 void RegisterRegion() {
			DependencyObject targetElement = HostControl;
			if(targetElement.CheckAccess()) {
				LayoutGroup lg = targetElement as LayoutGroup;
				if(lg != null && RegionManager != null) 
                    RegionManager.Regions.Add(Region);
			}
		}

        DependencyObject IHostAwareRegionBehavior.HostControl {
            get { return HostControl; }
            set { HostControl = value; }
        }
    }
}

File: VB/PrismOnDXDocking.Infrastructure/RegionBehaviors/LayoutGroupRegionBehavior.vb
Imports System
Imports System.ComponentModel.Composition
Imports System.Windows
Imports DevExpress.Xpf.Docking
Imports Prism.Regions.Behaviors
Imports Prism.Regions

Namespace PrismOnDXDocking.Infrastructure.Behaviors
    <Export(GetType(LayoutGroupRegionBehavior)), PartCreationPolicy(CreationPolicy.NonShared)> _
    Public Class LayoutGroupRegionBehavior
        Inherits RegionBehavior
        Implements IHostAwareRegionBehavior
        <Import>
        Public Property _RegionManager() As IRegionManager
        Private _hostControl As DependencyObject
        Public Property HostControl() As DependencyObject Implements IHostAwareRegionBehavior.HostControl
            Get
                Return _hostControl
            End Get
            Set(ByVal value As DependencyObject)
                If IsAttached Then
                    Throw New InvalidOperationException()
                End If
                Me._hostControl = value
            End Set
        End Property
        Protected Overrides Sub OnAttach()
            RegisterRegion()
        End Sub
        Private Sub RegisterRegion()
            Dim targetElement As DependencyObject = HostControl
            If targetElement.CheckAccess() Then
                Dim lg As LayoutGroup = TryCast(targetElement, LayoutGroup)
                If lg IsNot Nothing Then
                    If _RegionManager IsNot Nothing Then
                        _RegionManager.Regions.Add(Region)
                    End If
                End If
            End If
        End Sub
    End Class
End Namespace
